hypothesis-python/src/hypothesis/searchstrategy/strategies.py

```diff
@@ -480,14 +493,10 @@ def calc_label(self):
 
     def do_draw(self, data):
         # type: (ConjectureData) -> Ex
-        n = len(self.element_strategies)
-        assert n > 0
-        if n == 1:
-            return data.draw(self.element_strategies[0])
-
-        i = cu.integer_range(data, 0, n - 1)
+        assert len(self.element_strategies) > 0
 
-        return data.draw(self.element_strategies[i], label=self.branch_labels[i])
+        strategy = st.sampled_from(self.element_strategies).filter(lambda s: s.available(data)).flatmap(lambda s: s)
```

Comment:
```python
strategy = data.draw(SampledFromStrategy(self.element_strategies).filter(lambda s: s.available(data)))
```

Note that you'll need to run the autoformatters (`black` and `autoflake8`), and I'd drop the blank line and assertion above as it's covered by attempting to draw.

Reply:
90f6d13d62186adcc277e084a21607c7a13caee5